auth-web/src/components/datatable/BaseVDataTable.vue

```diff
@@ -138,63 +139,42 @@ export default defineComponent({
-    const timeout = ref(null)
-    const vm = getCurrentInstance().proxy
-    const regTable = ref(null)
-
-    const onScroll = (e) => {
-      timeout && clearTimeout(timeout)
-      timeout.value = setTimeout(() => {
-        const scrollTop = e.target.scrollTop
-        const rowsInView = Math.ceil(e.target.clientHeight / rowHeight.value)
-        let startRow = Math.floor(scrollTop / rowHeight.value)
-        const endRow = startRow + rowsInView
-
-        // Handle scrolling up
-        if (startRow < start.value) {
-          start.value = startRow
-        }
-
-        // Handle scrolling down
-        if (endRow > start.value + perPage) {
-          start.value = endRow - perPage
+    const currentPage = ref(1)
+    const perPage = ref(50)
+    const firstItem = ref(null) // first item in table
+    const reachedEnd = ref(false)
+
+    const getNext = _.throttle(() => {
+      // if not loading and reg history exists
+      if (!props.loading && !reachedEnd.value && state.sortedItems.length > state.visibleItems.length) {
+        currentPage.value++
+        const start = (currentPage.value - 1) * perPage.value
+        const end = start + perPage.value
+        state.visibleItems.push(...state.sortedItems.slice(start, end))
+        if (state.sortedItems.length <= state.visibleItems.length) {
+          reachedEnd.value = true
         }
+      }
+    }, 50, { trailing: false })
 
-        // Update the scrollTop value to match the new start row
-        vm.$nextTick(() => {
-          e.target.scrollTop = scrollTop
-        })
-      }, 20)
+    const scrollToTop = () => {
+      const table = document.querySelector('.v-data-table')
+      if (table) table.scrollTop = 0
     }
 
-    onMounted(() => {
-      const tableWrapper = document.querySelector('.v-data-table__wrapper')
-      if (tableWrapper) tableWrapper.addEventListener('scroll', onScroll)
-    })
-
-    const sortedItemsLimited = computed(() => {
-      const tableWrapper = document.querySelector('.v-data-table__wrapper')
-      if (tableWrapper) {
-        const rows = tableWrapper.getElementsByClassName('base-table__item-row')
-        if (rows.length > 0) rowHeight.value = Array.from(rows).reduce((acc, row) => acc + row.clientHeight, 0) / rows.length
+    watch(() => state.sortedItems, () => {
+      if (props.setItems && props.pageHide) {
+        state.visibleItems = state.sortedItems.slice(0, perPage.value)
+        firstItem.value = state.visibleItems[0]
+        currentPage.value = 1
+        reachedEnd.value = false
+        scrollToTop()
       }
-      return state.sortedItems.slice(start.value, perPage + start.value)
-    })
-
-    const startHeight = computed(() => {
-      return start.value * rowHeight.value - 80
-    })
-
-    const endHeight = computed(() => {
-      return rowHeight.value * (state.sortedItems.length - start.value)
-    })
+    }, { immediate: true })
 
     const filter = _.debounce(async (header: BaseTableHeaderI) => {
```

Comment:
Remove this debounce? or at least have a parameter to skip it? The table is way more responsive without the 500ms delay